src/devices/Bmp180/Bmp180.cs

```diff
@@ -0,0 +1,303 @@
+            }
+            else
+            {               
+                p = (int)(((long)B7 / B4) * 2);             
+            }
+            
+            X1 = (p / (1 << 8)) * (p / (1 << 8));
+            X1 = (X1 * 3038) / (1 << 16);
+            X2 = (-7357 * p) / (1 << 16);
+            p = p + (X1 + X2 + 3791) / (1 << 4);
+
+            return p;
+        }
+
+        /// <summary>
+        ///  Calculates the altitude in meters from the specified sea-level pressure(in hPa).
+        /// </summary>
+        /// <param name="seaLevelPressure" > 
+        ///  Sea-level pressure in hPa
+        /// </param>
+        /// <returns>
+        ///  Height in meters from the sensor
+        /// </returns>
+        public double ReadAltitude(double sealevel_pa = 101325.0)
+        {
+            double pressure = (double)ReadPressure();
+            double altitude = 44330.0 * (1.0 - Math.Pow((pressure / sealevel_pa), (1.0 / 5.255)));
+
+
+            return altitude;
+        }
+
+        /// <summary>
+        ///  Calculates the pressure at sealevel when given a known altitude in meter
+        /// </summary>
+        /// <param name="altitude" > 
+        ///  altitude in meters
+        /// </param>
+        /// <returns>
+        ///  Pressure in Pascals
+        /// </returns>
+        public double ReadSeaLevelPressure(double altitude = 0.0)
+        {
+            double pressure = (double)ReadPressure();
+            double p0 = pressure / Math.Pow((1.0 - (altitude / 44333.0)), 5.255);
+
+            return p0;
+        }
+
+        /// <summary>
+        ///  Reads raw temperatue from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Raw temperature
+        /// </returns>
+        short ReadRawTemperature()
+        {
+            // Reads the raw (uncompensated) temperature from the sensor
+            _i2cDevice.Write(new[] { (byte)Register.CONTROL, (byte)Register.READTEMPCMD });            
+            // Wait 5ms
+            Thread.Sleep(5);
+            short raw = (short)Read16BitsFromRegisterBE((byte)Register.TEMPDATA);
+            
+            return raw;
+        }
+
+
+        /// <summary>
+        ///  Reads raw pressure from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Raw pressure
+        /// </returns>
+        int ReadRawPressure()
+        {
+            // Reads the raw (uncompensated) pressure level from the sensor.
+            _i2cDevice.Write(new[] { (byte)Register.CONTROL, (byte)(Register.READPRESSURECMD + ((byte)Sampling.Standard << 6))});
+
+            if (this._mode.Equals(Sampling.UltraLowPower))
+            {
+                Thread.Sleep(5);
+            }else if (this._mode.Equals(Sampling.HighResolution))
+            {
+                Thread.Sleep(14);
+            }
+            else if (this._mode.Equals(Sampling.UltraHighResolution))
+            {
+                Thread.Sleep(26);

```

Comment:
Can you please explain these constants? If you use named constants instead, the code becomes self documenting.

Reply:
These constants are the conversion time depending on the sampling mode.

![time](https://user-images.githubusercontent.com/14133711/52922182-be691200-32e4-11e9-8f8f-1f5a342a282f.jpg)

Reply:
@garciaolais please add a comment that 26 is taken from the table in datasheet Page 12 or something like that

Reply:
Section number might be helpful too.